### DISCUSSION TITLE
IMPORT .gpg file

### DISCUSSION THREAD

**Discussion by ribbbo** [filamentphp/filament#6835] [Help]
any tips how to import .gpg file?

**Comment by lucacastelnuovo** [comment]
Could you elaborate? Importing a GPG file to where? With what purpose?

**Reply by ribbbo** [reply]
like importing a csv file to add in your data

**Reply by lucacastelnuovo** [reply]
What information of the GPG key do you want to store?
Where do you want to store it?
Do you just want to upload the file? (https://filamentphp.com/docs/2.x/forms/fields#file-upload)

**Comment by pboivin** [comment]
@ribbbo If it's not something that corresponds to a property on a model, I think you can still use the FileUpload field inside of a modal action custom form : https://filamentphp.com/docs/2.x/tables/actions#custom-forms

**Reply by ribbbo** [reply]
Yes, I'm currently using the FileUpload field. But the is save with the .bin extension. 
Any tips to make it save as .gpg or .txt?
The format of the file that I upload is filename.csv.gpg.

**Reply by lucacastelnuovo** [reply]
What are you storing? An encrypted CSV?

**Reply by lucacastelnuovo** [reply]
Assuming you are storing an encrypted CSV, and it gets renamed to .bin. Have you tried downloading the bin and decrypting the .bin file?

**Reply by ribbbo** [reply]
Yes I upload an encrypted CSV, How can I decrypt the .bin file?

**Reply by lucacastelnuovo** [reply]
Try `gpg --decrypt  filename.csv.bin`

**Reply by ribbbo** [reply]
In SSH?

**Reply by lucacastelnuovo** [reply]
You should enter the command at the location of your file. (or point to your file `gpg --decrypt  /some/directory/filename.csv.bin`)

Let's circle back to the beginning. I get the feeling that there are some basics you are missing.

The process so far:
1. You have an encrypted CSV. (How is it encrypted?)
2. You store this in a filament application. (What is your current upload code?)
3. When you retrieve this file, the name has changed. (What is your current download code?)
4. What are you going to do with the downloaded file?

**Reply by ribbbo** [reply]
My current process:

1.  I have a file named "filename.csv.gpg"
2.  I uploaded it in the filament admin. The code is:
Action::make('Import')
            ->form([
                FileUpload::make('import')
                ->disk('filament_gpg')
3. And the file was save as fMzsgNkIHR99oF2xOXZjezjbQKuycS-metaU0lUIElDIEJBU0UgMDkzMC5jc3YuZ3Bn-.bin

and after that I need to decrypt the uploaded file to store it in database.

**Reply by lucacastelnuovo** [reply]
What do you mean by `to store it in database`? 

Do you want to store the CSV data in the database? Do you want to store the file in the database?
How is the file encrypted, using symmetric or asymmetric encryption?

**Reply by ribbbo** [reply]
Store the csv data in database. 
I'm not sure how's the file encrypted

**Reply by lucacastelnuovo** [reply]
How were you planning to decrypt the file?

**Reply by ribbbo** [reply]
I have a source code about that using gnupg_init.

**Reply by lucacastelnuovo** [reply]
Okay, have you tried using that source code to decrypt `fMzsgNkIHR99oF2xOXZjezjbQKuycS-metaU0lUIElDIEJBU0UgMDkzMC5jc3YuZ3Bn-.bin`?
